### PR TITLE
added fix for user resource for 4.6

### PR DESCRIPTION
**Pull request by shenda1**
# Description
PR is raised to add support for first name/last name for user resource in PowerFlex 4.6

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- Acceptance Test
- Functional Test

### CODE REVIEW

**Comment by shenda1** [comment]
run e2e test

File: go.sum [hunk 1]
```diff
@@ -43,8 +43,12 @@ github.com/cyphar/filepath-securejoin v0.2.4/go.mod h1:aPGpWjXOXUn2NCNjFvBE6aRxG
 github.com/davecgh/go-spew v1.1.0/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
 github.com/davecgh/go-spew v1.1.1 h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c=
 github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
-github.com/dell/goscaleio v1.14.2-0.20240603053305-6eaea85323c8 h1:ulv4lMmp2+D4uhGbo9jfjmTHsWfRPPfy5wfrZLLXCnk=
-github.com/dell/goscaleio v1.14.2-0.20240603053305-6eaea85323c8/go.mod h1:h7SCmReARG/szFWBMQGETGkZObknhS45lQipQbtdmJ8=
+github.com/dell/goscaleio v1.14.2-0.20240527062756-acc6ecf03b1b h1:wNO2JHeijw2MAmLKXcqSLSTyD0y2V3/IdSrBDqjVlBM=
+github.com/dell/goscaleio v1.14.2-0.20240527062756-acc6ecf03b1b/go.mod h1:h7SCmReARG/szFWBMQGETGkZObknhS45lQipQbtdmJ8=
+github.com/dell/goscaleio v1.14.2-0.20240603144059-9e975a9edf84 h1:qNIdgiUYkwLkLYhgnkCvtvGMVQZrljFQDUHmDHKRLzU=
```
Comment (Krunal-Thakkar) [inline go.sum:48]: please run go mod tidy.

File: powerflex/provider/user_resource.go [hunk 2]
```diff
@@ -120,6 +120,24 @@ func (r *userResource) Schema(_ context.Context, _ resource.SchemaRequest, resp
 				},
 				MarkdownDescription: "Password of the user. For PowerFlex version 3.6, cannot be updated.",
 			},
+			"first_name": schema.StringAttribute{
+				Description:         "First name of the user. PowerFlex version 3.6 does not support the first_name attribute.",
+				Optional:            true,
+				Computed:            true,
+				MarkdownDescription: "First name of the user. PowerFlex version 3.6 does not support the first_name attribute.",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
```
Comment (Krunal-Thakkar) [inline powerflex/provider/user_resource.go:129]: Please add some validation here, like length min and max

Comment (shenda1) [reply powerflex/provider/user_resource.go:129]: For min length, API validation is there. For max length, I have tested with more than 128 chars, it's working fine so I am not sure whats the max length allowed.

File: powerflex/provider/user_resource.go [hunk 3]
```diff
@@ -160,6 +178,13 @@ func (r *userResource) ValidateConfig(plan models.UserModel) diag.Diagnostics {
 	if r.version == models.Version3X {
 		roles := roleMap[models.Version3X]
 
+		if !plan.FirstName.IsUnknown() || !plan.LastName.IsUnknown() {
+			diags.AddError(
+				"PowerFlex version 3.6 does not support the first_name and last_name attributes.",
+				"PowerFlex version 3.6 does not support the first_name and last_name attributes.",
```
Comment (AnikaAgiwal2711) [inline powerflex/provider/user_resource.go:184]: In 4.5 also it's not supported . Right ??

Comment (shenda1) [reply powerflex/provider/user_resource.go:184]: In 4.5, these params are optional whereas in 4.6, these params are mandatory.

**Review by AnikaAgiwal2711 — APPROVED** [review]

**Comment by shenda1** [comment]
run e2e test

**Review by rounak-adhikary — APPROVED** [review]

**Pull request merged**